diff --git a/CMakeLists.txt b/CMakeLists.txt
index 8581b4c..7ed5275 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -379,12 +379,6 @@ if (NOT CUSTOM_BLAS)
 endif (NOT CUSTOM_BLAS)
 
 if (OPENMP)
-  # Clang does not (yet) support OpenMP.
-  if (CMAKE_CXX_COMPILER_ID STREQUAL "Clang")
-    update_cache_variable(OPENMP OFF)
-    message("-- Compiler is Clang, disabling OpenMP.")
-    list(APPEND CERES_COMPILE_OPTIONS CERES_NO_THREADS)
-  else (CMAKE_CXX_COMPILER_ID STREQUAL "Clang")
     # Find quietly s/t as we can continue without OpenMP if it is not found.
     find_package(OpenMP QUIET)
     if (OPENMP_FOUND)
@@ -403,7 +397,6 @@ if (OPENMP)
       update_cache_variable(OPENMP OFF)
       list(APPEND CERES_COMPILE_OPTIONS CERES_NO_THREADS)
     endif (OPENMP_FOUND)
-  endif (CMAKE_CXX_COMPILER_ID STREQUAL "Clang")
 else (OPENMP)
   message("-- Building without OpenMP (disabling multithreading).")
   list(APPEND CERES_COMPILE_OPTIONS CERES_NO_THREADS)
@@ -632,7 +625,7 @@ endif (MSVC)
 if (UNIX)
   # GCC is not strict enough by default, so enable most of the warnings.
   set(CMAKE_CXX_FLAGS
-    "${CMAKE_CXX_FLAGS} -Werror=all -Werror=extra -Wno-unknown-pragmas -Wno-sign-compare -Wno-unused-parameter -Wno-missing-field-initializers")
+    "${CMAKE_CXX_FLAGS} -Wall -Wextra -Wno-unknown-pragmas -Wno-sign-compare -Wno-unused-parameter -Wno-missing-field-initializers")
 endif (UNIX)
 
 # Use a larger inlining threshold for Clang, since it hobbles Eigen,
diff --git a/internal/ceres/CMakeLists.txt b/internal/ceres/CMakeLists.txt
index 3a51309..3450449 100644
--- a/internal/ceres/CMakeLists.txt
+++ b/internal/ceres/CMakeLists.txt
@@ -163,7 +163,6 @@ endif (BLAS_FOUND AND LAPACK_FOUND)
 
 if (OPENMP_FOUND)
   if (NOT MSVC)
-    list(APPEND CERES_LIBRARY_PRIVATE_DEPENDENCIES gomp)
     list(APPEND CERES_LIBRARY_PRIVATE_DEPENDENCIES ${CMAKE_THREAD_LIBS_INIT})
   endif (NOT MSVC)
 endif (OPENMP_FOUND)
